### PR TITLE
Added symfony mailer, removed swift mailer

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

File: config/packages/dev/mailer.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+framework:
+    mailer:
+        # dsn: 'null://null'
+        envelope:
+            recipients: ['praneeth.pondicherry@tamucc.edu']
```
Comment (praneethpr) [inline config/packages/dev/mailer.yaml:5]: No new line, remove test code

File: src/Util/MailSender.php [hunk 2]
```diff
@@ -76,23 +85,15 @@ public function sendEmailMessage(
...
-            ->addPart($emailTwigTemplate->renderBlock('body_text', $mailData), 'text/plain');
-        foreach ($attachments as $attachment) {
-            if (!$attachment instanceof \Swift_Attachment) {
-                throw new \InvalidArgumentException('Attachment is not an instance of Swift_Attachment');
-            }
-            $message->attach($attachment);
-        }
+        $email = (new Email())
+            ->from(new Address($this->fromAddress, $this->fromName))
+            ->to(...$toAddresses)
```
Comment (praneethpr) [inline src/Util/MailSender.php:91]: Do we need name for $toAddress as well?

File: config/packages/dev/mailer.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+framework:
```
Comment (praneethpr) [inline config/packages/dev/mailer.yaml:1]: Do we need to create one for test as well? so that all emails goto a single email address?

Comment (mickel1138) [reply config/packages/dev/mailer.yaml:1]: I'm not sure if this config will use that new style yaml file. But if there is no file in test, the default one is used.

File: docker-compose.override.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+version: '3'
+
+services:
+###> symfony/mailer ###
+  mailer:
+    image: schickling/mailcatcher
+    ports: [1025, 1080]
+###< symfony/mailer ###
```
Comment (mickel1138) [inline docker-compose.override.yml:8]: I wonder what this file is.

File: src/Command/ValidateRemotelyHostedLinksCommand.php [hunk 5]
```diff
@@ -131,7 +131,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $this->mailer->sendEmailMessage(
                 $this->twig->load('Email/data-repository-managers.error-remotely-hosted.email.twig'),
                 array('errors' => $errors),
-                array('griidc@gomri.org' => 'GRIIDC')
+                array('griidc@gomri.org')
```
Comment (mickel1138) [inline src/Command/ValidateRemotelyHostedLinksCommand.php:134]: Here we may have to pass in a recipient instead. It called `Address`.

File: src/Event/EventListener.php [hunk 6]
```diff
@@ -151,8 +149,7 @@ protected function sendMailMsg(
             $this->mailer->sendEmailMessage(
                 $twigTemplate,
                 $mailData,
-                array($person->getEmailAddress() => $person->getFirstName() . ' ' . $person->getLastName()),
-                $attachments
+                array($person->getEmailAddress()),
```
Comment (mickel1138) [inline src/Event/EventListener.php:152]: Same here use, `Address` instead.

File: config/packages/dev/mailer.yaml (hunk 3 above)
Comment (praneethpr) [reply config/packages/dev/mailer.yaml:1]: Okay

File: src/Util/MailSender.php [hunk 7]
```diff
@@ -41,22 +50,23 @@ class MailSender
...
         TwigEnvironment $twig,
         string $fromAddress,
         string $fromName,
         string $bccAddress
     ) {
         $this->twig = $twig;
         $this->mailer = $mailer;
-        $this->from = array($fromAddress => $fromName);
+        $this->fromAddress = $fromAddress;
+        $this->fromName = $fromName;
```
Comment (mickel1138) [inline src/Util/MailSender.php:69]: How about we pass in Address instead. And type hint it like that also.

File: src/Util/MailSender.php (hunk 2 above)
Comment (mickel1138) [reply src/Util/MailSender.php:91]: yes

File: src/Command/AccountExpirationNotifyCommand.php [hunk 8]
```diff
@@ -251,7 +252,7 @@ protected function notifyAccountHolders(array $accounts, \Twig\TemplateWrapper $
                 $emailTemplate,
                 $twigData,
                 array(
-                    $person->getEmailAddress() => $person->getFirstName() . ' ' . $person->getLastName(),
+                    new Address($person->getEmailAddress(), 'GRIIDC'),
```
Comment (mickel1138) [inline src/Command/AccountExpirationNotifyCommand.php:255]: use names

**Review by mickel1138 — DISMISSED** [review]
LGTM

**Review by mickel1138 — APPROVED** [review]

**Pull request merged**